src/Symfony/Component/Messenger/Transport/Schedule/PeriodicalJob.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule;
+
+final class PeriodicalJob implements JobScheduleInterface
+{
+    public function __construct(
+        private readonly object $job,
+        private readonly int $intervalInSeconds,
+        private readonly \DateTimeImmutable $firstRun,
+        private readonly \DateTimeImmutable $priorTo,
+    ) {
+    }
+
+    public static function create(
+        object $job,
+        string|int|\DateInterval $interval,
+        string|\DateTimeImmutable $firstRun,
+        string|\DateTimeImmutable $priorTo = Schedule::FAR_FUTURE,
+    ): self {
+        if (\is_string($firstRun)) {
+            $firstRun = new \DateTimeImmutable($firstRun);
+        }
+        if (\is_string($priorTo)) {
+            $priorTo = new \DateTimeImmutable($priorTo);
+        }
+        if (\is_string($interval)) {
+            if ('P' === $interval[0]) {
+                $interval = new \DateInterval($interval);
+            } else {
+                \assert((string) (int) $interval === $interval);
```

Comment:
Is it really useful?

Reply:
Oh... I will change it to `ctype_digit`, it would be more clear. It's a check that string contains integers only.